Sources/SwiftUIKit/Views/CollectionView.swift

```diff
@@ -525,3 +539,126 @@ extension CollectionView: UICollectionViewDelegate {
+        
+        if shouldUpdate {
+            layoutIfNeeded()
+        }
+        
+        return self
+    }
+    
+    @discardableResult
+    func minimumLineSpacing(shouldUpdate: Bool = true, forSections spacings: [CGFloat]) -> Self {
+        minimumLineSpacingForSections = spacings
+        
+        if shouldUpdate {
+            layoutIfNeeded()
+        }
+        
+        return self
+    }
+    
+    @discardableResult
+    func minimumInteritemSpacing(shouldUpdate: Bool = true, forSections spacings: [CGFloat]) -> Self {
+        minimumInteritemSpacingForSections = spacings
+        
+        if shouldUpdate {
+            layoutIfNeeded()
+        }
+        
+        return self
+    }
+    
+    @discardableResult
+    func headerSize(shouldUpdate: Bool = true, forSections sizes: [CGSize]) -> Self {
+        headerSizesForSections = sizes
+        
+        
+        if shouldUpdate {
+            layoutIfNeeded()
+        }
+        
+        return self
+    }
+    
+    @discardableResult
+    func footerSize(shouldUpdate: Bool = true, forSections sizes: [CGSize]) -> Self {
+        footerSizeForSections = sizes
+        
+        if shouldUpdate {
+            layoutIfNeeded()
+        }
+        
+        return self
+    }
+}
+
+// MARK: - Collection Flow Layout Delegate
+@available(iOS 11, *)
+extension CollectionView: UICollectionViewDelegateFlowLayout {
+    func returnValidValue<T>(for array: [T]?, section: Int) -> T? {
+            if let array = array,
+            array.count != 0 {
+            
+            if array.count != numberOfSections {
+                let element = array[0]
+                return element
+            } else {
+                let element = array[section]
+                return element
+            }
+        }
+        
+        return nil
+    }
+    
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        data[indexPath.section][indexPath.row].bounds.size
+    }
+    
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        if let inset = returnValidValue(for: sectionsInsets, section: section) {
+            return inset
+        }
+        
+        return UIEdgeInsets()
+    }
+    
+    
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        if let spacing = returnValidValue(for: minimumLineSpacingForSections, section: section) {
+            return spacing
+        }
+        
+        return 10
+    }
+    
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+        
```

Comment:
🤔 

```swift
if let spacing = returnValidValue(for: minimumInteritemSpacingForSections, section: section) {
    return spacing
}
```

Reply:
That's a method for handling given amount of spacings/anything in this extension - if array's count isn't equal to number of sections in collection view, it returns first value of that array. If it is equal, it returns size for given section index. If method unwrapped a nil array, it returns nil as well, so on the end I am adding default value in delegate method.

I should document that and maybe rename it to use more meaningful name.

If we don't want to handle that in this way, it won't be a problem to change it to something else.

Reply:
@Oschly I am fine with this. I was just pointing out this function doesn't have that bit of code I expect... It just returns 5 always. 

`minimumInteritemSpacingForSections` should be used in this function.